File: Advanced Methods of Programming/JavaClientServer/Model/src/main/java/Model/Proba.java
package Model;

import java.io.Serializable;

public class Proba implements HasId<Integer>, Serializable {
    private int id;
    private Distanta distanta;
    private Stil stil;

    public Proba(Integer idProba, Integer distanta, String stil){
        this.id=idProba;
        this.stil=Stil.valueOf(stil);
        this.distanta=Distanta.valueOf(Distanta.getValue(distanta));
    }
    public Proba(){}

    @Override
    public Integer getId() {
        return id;
    }

    @Override
    public void setId(Integer integer) { this.id=integer; }

    public Distanta getDistanta() {
        return distanta;
    }

    public void setDistanta(Distanta distanta) {
        this.distanta = distanta;
    }

    public Stil getStil() {
        return stil;
    }

    public void setStil(Stil stil) {
        this.stil = stil;
    }

    public Integer distantaInt(){
        return Distanta.getValueInt(this.distanta.toString());
    }
}

File: Advanced Methods of Programming/Client-ServerCSharp/SharedDLL/Domain/Proba.cs
using System;

namespace SharedDLL.Domain
{
    [Serializable]
    public class Proba : IHasID<int>
    {
        private int IdProba;
        private string Stil;
        private int Distanta;
       public Proba(int Id, string St, int Dist)
        {
            this.IdProba = Id;
            this.Stil = St;
            this.Distanta = Dist;
        }

        public int GetId() { return this.IdProba; }

        public string GetStil() { return this.Stil; }

        public int GetDistanta() { return this.Distanta; }
    }
}
